src/scripts/main.js

```diff
@@ -1 +1,67 @@
 'use strict';
+
+function createMessage(message, type, posTop, posRight) {
+  const div = document.createElement('div');
+
+  div.innerHTML = `
+    <div data-qa="notification" class="message ${type}">
+      ${message}
+    </div>
+  `;
+
+  div.style.position = 'absolute';
+  div.style.top = posTop + 'px';
+  div.style.right = posRight + 'px';
+  document.body.append(div);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+
+  document.addEventListener('click', () => {
+    clearTimeout(timeoutId);
+    resolve('First promise was resolved');
+  });
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    resolve('Second promise was resolved');
+  });
```

Comment:
I guess in this case better use mousedown and check `e.button` instead of 2 listeners